systems/plants/test/testManipulatorDynamics.m

```diff
@@ -0,0 +1,190 @@
+function testManipulatorDynamics
+testActuatedPendulum();
+regressionTestAtlasRPY();
+checkGradients(createAtlas('rpy'));
+checkHdotMinus2CoriolisMatrixSkewSymmetricMatrix(createAtlas('rpy'))
+
+if RigidBodyManipulator.use_new_kinsol
+  testBrickQuaternion();
+  testAtlasQuat();
+  checkGradients(createFallingBrick('quat'));
+  checkGradients(createAtlas('quat'));
+end
+end
+
+function robot = createFallingBrick(floating_type)
+options.floating = floating_type;
+robot = RigidBodyManipulator('FallingBrick.urdf',options);
+end
+
+function testBrickQuaternion()
+options.floating = 'quat';
+r = RigidBodyManipulator('FallingBrick.urdf',options);
+nv = r.getNumVelocities();
+q = getRandomConfiguration(r);
+v = randn(nv, 1);
+[H,C,B] = manipulatorDynamics(r,q,v,false);
+
+valuecheck(H, r.body(2).I, 1e-12);
+% C: Newton-Euler equations:
+twist_omega = v(1:3);
+twist_v = v(4:6);
+m = r.body(2).mass;
+I = r.body(2).inertia;
+gravity_body = quatRotateVec(quatConjugate(q(4:7)), r.gravity);
+C_check = [...
+  cross(twist_omega, I * twist_omega);
+  cross(twist_omega, m * twist_v) - m * gravity_body];
+valuecheck(C, C_check);
+valuecheck([nv 0], size(B));
+end
+
+function testActuatedPendulum()
+m = RigidBodyManipulator('ActuatedPendulum.urdf');
+nq = m.getNumPositions();
+nv = m.getNumVelocities();
+q = randn(nq, 1);
+v = randn(nv, 1);
+[H,C,B] = manipulatorDynamics(m,q,v,false);
+
+body = m.body(2);
+axis = body.joint_axis;
+I = body.I;
+H_expected = axis' * I(1:3, 1:3) * axis;
+valuecheck(H_expected, H, 1e-12);
+% TODO: check C
+valuecheck(1, B);
+
+checkMex(m);
+end
+
+function regressionTestAtlasRPY()
+replaceMatFile = false;
```

Comment:
The file is 21 kB large so I don't think it's a huge issue. Also, I need stored results to compare to if I want to do a regression test.